### PR TITLE
[Github Config] Launch testing jobs only with code changes 

### PR DESCRIPTION
**Pull request by danhoeflinger**
Skip testing for PRs which don't affect code / testing.

Require that they touch files in one of the following paths:
```
      - '.github/**/*'
      - 'cmake/**/*'
      - 'include/**/*'
      - 'test/**/*'
 ```
Always run codespell and documentation jobs.

### CODE REVIEW

File: .github/workflows/ci-testing.yml
Comment (danhoeflinger) [inline .github/workflows/ci-testing.yml:15]: This is the important addition, everything else is just separating / moving the jobs as is to a new workflow

**Comment by danhoeflinger** [comment]
Please see https://github.com/oneapi-src/oneDPL/actions/runs/11918913730
and https://github.com/oneapi-src/oneDPL/actions 
To see that before the `.github` folder was added to the paths list, that only the documentation workflow was launched (codespell and documentation), once I added the .github folder to the path filter, all jobs were launched (see current CI actions).

File: .github/workflows/ci-testing.yml
Comment (MikeDvorskiy) [reply .github/workflows/ci-testing.yml:15]: I guess, we should add `examples `and `make` as well.

Comment (MikeDvorskiy) [reply .github/workflows/ci-testing.yml:15]: 1) about make:
I don't know reasons why 'make'  would not work.
Make configuration still works, I use it.
But if CI doesn't use make files, we can exclude that folder, of course.

2) "we don't test the examples in the CI."..  
Interesting, it was done intentionally to minimize `CI per commit` time or just we forgot about that?

Comment (danhoeflinger) [reply .github/workflows/ci-testing.yml:15]: > 1. about make:
>    I don't know reasons why 'make'  would not work.
>    Make configuration still works, I use it.
>    But if CI doesn't use make files, we can exclude that folder, of course.

As far as I understand, the CI generates "fresh" build configuration using `cmake`, then uses that build configuration to build the tests.  Of course, there is nothing wrong with using the make config if it works, it just has not changed much in the last few years and I don't actively use it.  So I'm not aware of its current support for all cases.

> 2. "we don't test the examples in the CI."..
>    Interesting, it was done intentionally to minimize `CI per commit` time or just we forgot about that?

Its possible we build and run the examples and I am unaware of it.  I can try to look into this.  It would be a good idea to do if we aren't already.  I suppose we can add it in here in anticipation of adding these to the CI in the future.

Comment (danhoeflinger) [reply .github/workflows/ci-testing.yml:15]: Good point about `CMakeLists.txt` ... 
Let me add that.  
I'm OK with leaving out `examples` for now, and adding it if we add it in the future to the CI.

File: .github/workflows/ci.yml
Comment (danhoeflinger) [reply .github/workflows/ci.yml]: Updated this.

File: .github/workflows/ci-testing.yml
Comment (danhoeflinger) [reply .github/workflows/ci-testing.yml]: Cleaned these, and actually all the env variables are unused in the docs so I removed them

**Comment by danhoeflinger** [comment]
@timmiesmith I updated the badge support on the Readme.md to use the badge from the testing workflow.  Do you want me to add one for the docs as well?  I don't want to clutter it.

**Review by dmitriy-sobolev — APPROVED** [review]
LGTM. Could you wait for the CI completion before merging?

**Pull request merged**
